### DISCUSSION TITLE
Flutter Firebase App Check

### DISCUSSION THREAD

**Discussion by febg11** [firebase/flutterfire#6276] [Q&A]
I would like to intergrate this into my app but a have a couple questions

1) The usage for safetynet is 10k per day. How often does the firebase_app_check perform a request? Is it everytime the app is launched.

2) If i reach the 10k threshold what exceptions are thrown and how can I handle them?

Thanks

**Comment by noahabe** [comment]
To be honest, I am not a firebase_app_check expert, so the following answers are subjected to errors. 

To answer your first question, it depends on the TTL (Time To Live) you set in Firebase App Check. I quote the following from the [documentation](https://firebase.google.com/docs/app-check/flutter/default-providers#:~:text=Note%20that%20the%20App%20Check%20library%20refreshes%20tokens%20at%20approximately%20half%20the%20TTL%20duration.) 

"The default TTL is reasonable for most apps. Note that the App Check library refreshes tokens at approximately half the TTL duration."

So it is not making a request every time the app is launched. It only makes a request when half the TTL is reached.

**Comment by gOzaru** [comment]
You can request to raise up the quota and the official documentation of Firebase App Check. If your use case is considered to be important and necessary, they will raise it up.
